core/src/main/java/org/testcontainers/utility/TestcontainersConfiguration.java

```diff
@@ -238,4 +157,19 @@ private static Properties readProperties(URL url) {
         }
         return properties;
     }
+
+    DockerImageName getConfiguredSubstituteImage(DockerImageName original) {
+        for (final Map.Entry<DockerImageName, String> entry : CONTAINER_MAPPING.entrySet()) {
+            if (original.isCompatibleWith(entry.getKey())) {
```

Comment:
Risk: if so-called `tiny image` (alpine:3.5) is overridden in config, all usages of that image will be replaced. Substitution is now along image-identity lines rather than semantic lines.